sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsServiceSerializer.cs

```diff
@@ -0,0 +1,776 @@
+            }
+
+            return default;
+        }
+
+        private static void ReadDocumentErrors(JsonElement documentElement, List<DocumentError> errors)
+        {
+            if (documentElement.TryGetProperty("errors", out JsonElement errorsValue))
+            {
+                foreach (JsonElement errorElement in errorsValue.EnumerateArray())
+                {
+                    DocumentError error = new DocumentError();
+
+                    if (errorElement.TryGetProperty("id", out JsonElement idValue))
+                        error.Id = idValue.ToString();
+                    if (errorElement.TryGetProperty("error", out JsonElement errorValue))
+                    {
+                        if (errorsValue.TryGetProperty("message", out JsonElement messageValue))
+                        {
+                            error.Message = messageValue.ToString();
+                        }
+                    }
+
+                    errors.Add(error);
+                }
+            }
+        }
+
+        private static string ReadModelVersion(JsonElement documentElement)
+        {
+            if (documentElement.TryGetProperty("modelVersion", out JsonElement modelVersionValue))
+            {
+                return modelVersionValue.ToString();
+            }
+
+            return default;
+        }
+
+        private static DocumentBatchStatistics ReadDocumentBatchStatistics(JsonElement documentElement)
+        {
+            if (documentElement.TryGetProperty("statistics", out JsonElement statisticsElement))
+            {
+                DocumentBatchStatistics statistics = new DocumentBatchStatistics();
+
+                if (statisticsElement.TryGetProperty("documentsCount", out JsonElement documentCountValue))
+                    statistics.DocumentCount = documentCountValue.GetInt32();
+                if (statisticsElement.TryGetProperty("validDocumentsCount", out JsonElement validDocumentCountValue))
+                    statistics.ValidDocumentCount = validDocumentCountValue.GetInt32();
+                if (statisticsElement.TryGetProperty("erroneousDocumentsCount", out JsonElement erroneousDocumentCountValue))
+                    statistics.ErroneousDocumentCount = erroneousDocumentCountValue.GetInt32();
+                if (statisticsElement.TryGetProperty("transactionsCount", out JsonElement transactionCountValue))
+                    statistics.DocumentCount = transactionCountValue.GetInt32();
+
+                return statistics;
+            }
+
+            return default;
+        }
+
+        #endregion Deserialize Common
+
+        #region Detect Languages
+
+        public static async Task<DocumentResultCollection<DetectedLanguage>> DeserializeDetectLanguageResponseAsync(Stream content, CancellationToken cancellation)
+        {
+            using (JsonDocument json = await JsonDocument.ParseAsync(content, cancellationToken: cancellation).ConfigureAwait(false))
```

Comment:
Any reason for the methods below to not use a `using` variable that is scoped to the method instead of an explicit scope (like the `using DiagnosticScope scope = _clientDiagnostics.CreateScope("Azure.AI.TextAnalytics.TextAnalyticsClient.ExtractEntityLinking");` declarations)?